CMakeLists.txt

```diff
@@ -1,6 +1,20 @@
 cmake_minimum_required(VERSION 3.22)
 project(ulight)
 
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "arm64")
+    set(HOMEBREW_PREFIX "/opt/homebrew") # Silicon
+else()
+    set(HOMEBREW_PREFIX "/usr/local") # Intel
+endif()
+
+if(APPLE)
+    set(CMAKE_EXE_LINKER_FLAGS "-L${HOMEBREW_PREFIX}/opt/llvm/lib")
+    set(CMAKE_MODULE_LINKER_FLAGS "-L${HOMEBREW_PREFIX}/opt/llvm/lib")
+    set(CMAKE_SHARED_LINKER_FLAGS "-L${HOMEBREW_PREFIX}/opt/llvm/lib")
+    set(CMAKE_AR "${HOMEBREW_PREFIX}/opt/llvm/bin/llvm-ar")
+    set(CMAKE_RANLIB "${HOMEBREW_PREFIX}/opt/llvm/bin/llvm-ranlib")
+endif()
+
```

Comment:
This should be all in a toolchain file or something, outside the project. The main CMakeLists shouldn't need to contain these system-specific setups, and I'm not sure if these paths are always the same, e.g. when you have multiple version of llvm installed on your system. The rest of the PR looks good.